cluster/ubuntu/util.sh

```diff
@@ -306,10 +306,7 @@ function kube-up() {
   source "${KUBE_ROOT}/cluster/ubuntu/${KUBE_CONFIG_FILE:-config-default.sh}"
 
   # downloading tarball release
-  if [[ -d "${KUBE_ROOT}/cluster/ubuntu/binaries" ]]; then
-    rm -rf "${KUBE_ROOT}/cluster/ubuntu/binaries"
-  fi
-  "${KUBE_ROOT}/cluster/ubuntu/download-release.sh"
+  [[ -d "${KUBE_ROOT}/cluster/ubuntu/binaries" ]] || "${KUBE_ROOT}/cluster/ubuntu/download-release.sh"
```

Comment:
Your code is *not* correct. If you insist, you should handle cross validation between local binaries and $KUBE_VERSION
Moreover you should take care of the networking issue yourself.